Home_Work_Term/Game.cpp

```diff
@@ -0,0 +1,246 @@
+#include "Game.h"
+#include <iostream>
+#include "Projectile.h"
+#include <cstdlib> 
+#include <random>
+
+Game::Game() 
+{
+	m_backTexture.loadFromFile("gameBack.jpg");
+	m_backSprite.setTexture(m_backTexture);
+
+	smallShipSpawnPeriod_Cycles = 5000;
+	smallShipSpawnCyclesPassed = 0;
+
+	m_battleMusic.openFromFile("battleMusic.wav");
+	m_battleMusic.setLoop(true);
+	m_battleMusic.setVolume(50);
+
+	m_GameOver.openFromFile("gameover.wav");
+	m_GameOver.setLoop(false);
+	m_GameOver.setVolume(50);
+
+}
+
+void Game::setplayersProjectiles(Projectile* playersProjectile)
+{
+	m_playersProjectiles.push_back(playersProjectile);
+}
+
+void Game::processPlayersProjectileFly()
+{
+	if (m_playersProjectiles.size() > 0) 
+	{
+		
+		auto it = m_playersProjectiles.begin();
+		while (it != m_playersProjectiles.end())
+		{
+			(*it)->fly();
+			if ((*it)->notOnTheScreen())
+			{
+				delete* it;
+				it = m_playersProjectiles.erase(it);
+			}
+			else
+			{
+				++it;
+			}
+		}
+	}
+}
+
+void Game::draw(sf::RenderWindow& window)
+{
+	if (m_playersProjectiles.size() > 0)
+	{
+		for (int i = 0; i < m_playersProjectiles.size(); i++)
+		{
+			m_playersProjectiles[i]->draw(window);
+		}
+	}
+
+	if (m_enemyShips.size() > 0)
+	{
+		for (int i = 0; i < m_enemyShips.size(); i++)
+		{
+			m_enemyShips[i]->draw(window);
+		}
+	}
+
+}
+
+void Game::generateEnemy()
+{
+	smallShipSpawnCyclesPassed++;
+	if (smallShipSpawnCyclesPassed>= smallShipSpawnPeriod_Cycles)
+	{
+		enemySpaceShip* generatedEnemy = new smallFastShip;
+
+		sf::VideoMode desktopMode = sf::VideoMode::getDesktopMode();
+		unsigned int screenWidth = desktopMode.width;
+
+		std::random_device rd;
+		std::mt19937 gen(rd());
+		std::uniform_int_distribution<int> distributionY(-650, -150);
+		std::uniform_int_distribution<int> distributionX(150, screenWidth-150);
```

Comment:
It would be better to move out all constants like here into Config class. You'll then could tweak your gameplay logic more easily from one place. Plausibly, later moving it into a disk file to make it possible to change UX without even needing to recompile the code.